packages/webpack/mixins/fastRefresh/mixin.core.js

```diff
@@ -0,0 +1,28 @@
+const { Mixin } = require('hops-mixin');
+const ReactRefreshWebpackPlugin = require('@pmmmwh/react-refresh-webpack-plugin');
+
+class ReactRefreshMixin extends Mixin {
+  configureBuild(webpackConfig, { jsLoaderConfig }, target) {
+    const { fastRefresh } = this.options;
+
+    if (target === 'develop' && fastRefresh) {
```

Comment:
Correct me if I'm wrong but with this conditional we only update frontend config right? only for frontend when we are developing and the flag is true.

Another thing that I want to ask if it make sense to have a new mixin (this one) or would be better to update only the `WebpackOptimizationsMixin`?

Reply:
Yes, that's correct.
I think it would make sense to put this into the optimisations mixin, because there's a code path that removes all babel plugins when TypeScript is being used.

When hops-typescript is installed, we add the babel loader with all of its options too, to be able to do automatic polyfilling and to make sure that `importComponent` works: https://github.com/xing/hops/blob/master/packages/typescript/mixin.core.js#L31-L45
But in the optimisations mixin, we remove all babel plugins except for our own `importComponent`: https://github.com/xing/hops/blob/master/packages/webpack/mixins/optimizations/mixin.core.js#L74-L88, so I think it makes sense to add fast-refresh there too.